File: launchdarkly/ruby/src/CreateDeploymentEventExample.rb
require "json"
require "launchdarkly_client"

LaunchDarklyClient.configure do |config|
    config.api_key["ApiKey"] = "YOUR_API_KEY"
end

post_deployment_event_input = LaunchDarklyClient::PostDeploymentEventInput.new
post_deployment_event_input.project_key = "default"
post_deployment_event_input.environment_key = "production"
post_deployment_event_input.application_key = "billing-service"
post_deployment_event_input.version = "a90a8a2"
post_deployment_event_input.event_type = "started"
post_deployment_event_input.application_name = "Billing Service"
post_deployment_event_input.application_kind = "server"
post_deployment_event_input.version_name = "v1.0.0"
post_deployment_event_input.event_time = 1706701522000
post_deployment_event_input.event_metadata = JSON.parse(<<-EOD
    {
        "buildSystemVersion": "v1.2.3"
    }
    EOD
)
post_deployment_event_input.deployment_metadata = JSON.parse(<<-EOD
    {
        "buildNumber": "1234"
    }
    EOD
)

begin
    LaunchDarklyClient::InsightsDeploymentsBetaApi.new.create_deployment_event(
        post_deployment_event_input,
    )
rescue LaunchDarklyClient::ApiError => e
    puts "Exception when calling InsightsDeploymentsBetaApi#create_deployment_event: #{e}"
end

File: launchdarkly/python/src/CreateDeploymentEventExample.py
import json
from datetime import date, datetime
from pprint import pprint

from launchdarkly_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"ApiKey": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    post_deployment_event_input = models.PostDeploymentEventInput(
        projectKey="default",
        environmentKey="production",
        applicationKey="billing-service",
        version="a90a8a2",
        eventType="started",
        applicationName="Billing Service",
        applicationKind="server",
        versionName="v1.0.0",
        eventTime=1706701522000,
        eventMetadata=json.loads("""
            {
                "buildSystemVersion": "v1.2.3"
            }
        """),
        deploymentMetadata=json.loads("""
            {
                "buildNumber": "1234"
            }
        """),
    )

    try:
        api.InsightsDeploymentsBetaApi(api_client).create_deployment_event(
            post_deployment_event_input=post_deployment_event_input,
        )
    except ApiException as e:
        print("Exception when calling InsightsDeploymentsBetaApi#create_deployment_event: %s\n" % e)
